test/telepilot_test.go

```diff
@@ -0,0 +1,161 @@
+package telepilot_test
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"os"
+	"path"
+	"testing"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.creack.net/telepilot/pkg/apiclient"
+	"go.creack.net/telepilot/pkg/apiserver"
+	"go.creack.net/telepilot/pkg/tlsconfig"
+)
+
+func noError(t *testing.T, err error, msg string) {
+	t.Helper()
+	if err != nil {
+		t.Fatalf("%s: %s.", msg, err)
+	}
+}
+
+func assert[T comparable](t *testing.T, expect, got T, msg string) {
+	t.Helper()
+	if expect != got {
+		t.Fatalf("Assert fail: %s:\n Expect:\t%v\n Got:\t%v", msg, expect, got)
+	}
+}
+
+func loadTLSConfig(t *testing.T, name string) *tls.Config {
+	t.Helper()
+	const certDir = "../certs"
+
+	// Make sure we have the expected key. Just check for one, assume that the ca and cert are present if the key is there.
+	if _, err := os.Stat(path.Join(certDir, name+".pem")); err != nil {
+		t.Skip("Missing cert, skipping. Make sure to run `make mtls` first or invoke tests with `make test`.")
+	}
+	cfg, err := tlsconfig.LoadTLSConfig(
+		path.Join(certDir, name+".pem"),
+		path.Join(certDir, name+"-key.pem"),
+		path.Join(certDir, "ca.pem"),
+		name != "server",
+	)
+	noError(t, err, "Load certs for "+name)
+	return cfg
+}
+
+func TestStartStop(t *testing.T) {
+	t.Parallel()
+
+	// Load certs for the server and a couple of clients.
+	serverTLSConfig := loadTLSConfig(t, "server")
+	aliceTLSConfig := loadTLSConfig(t, "client-alice")
+	bobTLSConfig := loadTLSConfig(t, "client-bob")
+
+	// Create a server.
+	srv := apiserver.NewServer(serverTLSConfig)
+
+	// Listen on a random port.
+	lis, err := net.Listen("tcp", "127.0.0.1:0")
+	noError(t, err, "Listen")
+	t.Cleanup(func() { _ = lis.Close() }) // No strictly needed, but just to make sure.
```

Comment:
I'd say that it's sort of needed if you ever want to run the test with `-count 10000` or so.

Reply:
I’ll update the comment, it is not needed because it will be invoked by the server close()